### ISSUE TITLE
MACServerExploit not working on rb450g ROS 6.39.3 Stable

### ISSUE THREAD

**Issue by puppet13th** [BasuCert/WinboxPoC#36]
i have rb450g with disabled winbox and ssh port access,so the only way to access the devices is via MAC address.Therefore i cant use WinboxExploit.py to get the forgotten admin password.when i run MACServerExploit.py it gave me error

on my windows 10 with WSL(debian):

> WinboxPoC$ python3 MACServerExploit.py 6c:3b:6b:XX:XX:XX
> Traceback (most recent call last):
>   File "MACServerExploit.py", line 160, in <module>
>     dump(m.recv(contains=b"\x11\x00\x00\x21"))
>   File "/home/WinboxPoC/extract_user.py", line 45, in dump
>     user_pass = get_pair(data)
>   File "/home/WinboxPoC/extract_user.py", line 38, in get_pair
>     user  = user.decode("ascii")
> UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
> ^CException ignored in: <module 'threading' from '/usr/lib/python3.5/threading.py'>
> Traceback (most recent call last):
>   File "/usr/lib/python3.5/threading.py", line 1288, in _shutdown
>     t.join()
>   File "/usr/lib/python3.5/threading.py", line 1054, in join
>     self._wait_for_tstate_lock()
>   File "/usr/lib/python3.5/threading.py", line 1070, in _wait_for_tstate_lock
>     elif lock.acquire(block, timeout):
> KeyboardInterrupt

on my raspberry pi 0 w raspbian buster :

> WinboxPoC $ python3 MACServerExploit.py 6c:3b:6b:XX:XX:XX
> Traceback (most recent call last):
>   File "MACServerExploit.py", line 160, in <module>
>     dump(m.recv(contains=b"\x11\x00\x00\x21"))
>   File "/home/pi/WinboxPoC/extract_user.py", line 45, in dump
>     user_pass = get_pair(data)
>   File "/home/pi/WinboxPoC/extract_user.py", line 38, in get_pair
>     user  = user.decode("ascii")
> UnicodeDecodeError: 'ascii' codec can't decode byte 0xff in position 0: ordinal not in range(128)
> ^CException ignored in: <module 'threading' from '/usr/lib/python3.7/threading.py'>
> Traceback (most recent call last):
>   File "/usr/lib/python3.7/threading.py", line 1281, in _shutdown
>     t.join()
>   File "/usr/lib/python3.7/threading.py", line 1032, in join
>     self._wait_for_tstate_lock()
>   File "/usr/lib/python3.7/threading.py", line 1048, in _wait_for_tstate_lock
>     elif lock.acquire(block, timeout):
> KeyboardInterrupt

i can confirm this exploit (WinboxExploit.py and MACServerExploit.py) works on RBmAPL-2nD ROS 6.39.2 Stable and 6.40 Stable.i have tried to disable winbox and ssh port on RBmAPL-2nD and the exploit still works.

any clue guys?

edit : webfig still available for access on rb450g

**Comment by rescenic** [comment]
Try adding "ignore" parameter in line 38 of "extract_user.py" file.
```
37 pass_plain = decrypt_password(user, pass_encrypted)
38 user  = user.decode("ascii","ignore")
```

**Comment by Deantwo** [comment]
Or see #46.

**Comment by mosajjal** [comment]
should be fixed with #46. Closing
